presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestApproximateCountDistinctBoolean.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static com.facebook.presto.metadata.FunctionKind.AGGREGATE;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+
+public class TestApproximateCountDistinctBoolean
+        extends AbstractTestApproximateCountDistinct
+{
+    @Override
+    public InternalAggregationFunction getAggregationFunction()
+    {
+        return metadata.getFunctionRegistry().getAggregateFunctionImplementation(
+                new Signature("approx_distinct", AGGREGATE, BIGINT.getTypeSignature(), BOOLEAN.getTypeSignature(), DOUBLE.getTypeSignature()));
+    }
+
+    @Override
+    public Type getValueType()
+    {
+        return BOOLEAN;
+    }
+
+    @Override
+    public Object randomValue()
+    {
+        return ThreadLocalRandom.current().nextBoolean();
+    }
+
+    @DataProvider(name = "inputSequences")
+    public Object[][] inputSequences()
+    {
+        return new Object[][] {
+                {true},
+                {false},
+                {true, false},
+                {true, true, true},
+                {false, false, false},
+                {true, false, true, false},
+        };
+    }
+
+    @Test(dataProvider = "inputSequences")
+    public void testNonEmptyInputs(boolean... inputSequence)
+    {
+        List<Object> booleanList = new ArrayList<>();
+        for (boolean b : inputSequence) {
+            booleanList.add(b);
+        }
+        assertCount(booleanList, 0, distinctCount(booleanList));
+    }
+
+    @Test
+    public void testNoInput()
+    {
+        List<Object> booleanList = new ArrayList<>();
+        assertCount(ImmutableList.of(), 0, 0);
+    }
+
+    private long distinctCount(List<Object> inputSequence)
```

Comment:
The only call site is `testNonEmptyInputs` where (after applying my comment above) you will know this is `List<Boolean>`. Thus the method can be simplified down to:

```
private long distinctCount(List<Boolean> inputSequence)
{
    return ImmutableSet.copyOf(inputSequence).size();
}
```

(with no need to check instanceof or cast, there is no need to use a Stream)

Reply:
thanks for the patience, doing this

> lly, you should squash the commits so that Github's merge button can be used.
> This is better than doing the squashin

ok, squashing it locally and then force pushing the branch

Reply:
squashed the commits